### ISSUE TITLE
Add predefined MimeType instances for the most common types

### ISSUE THREAD

**Issue by sa-bpelakh** [semanticarts/gist#463]
I suggest:
* `text/plain`, `text/html`, `text/csv`
* `application/json`, `application/ld+json`
* `application/rdf+xml`, `application/turtle`, `text/turtle`, `application/n-triples`
* `application/sparql-results+json`,`application/sparql-results+xml`

Need a good way to create local names for these, e.g.
```
gist:_MimeType_ApplicationJson
    a gist:MimeType ;
    skos:prefLabel "application/json" .
```

**Comment by rjyounes** [comment]
Aha! We are developing a set of conventions for that (as you know). Alphanumeric only, word boundaries (I.e., capital letter) at site of removed alphanum. So exactly what you show.

I like these.

**Comment by Jamie-SA** [comment]
URIs for the mime types already exist, but I haven't found a document of them so I created triples I needed as follows:
```
@prefix mime: <https://www.iana.org/assignments/media-types/> .
mime: {
  mime:text\/html rdf:type gist:MimeType
  ; skos:prefLabel "HTML"
  ; gist:uniqueText "text/html"
  .
  mime:text\/turtle rdf:type gist:MimeType
  ; skos:prefLabel "Turtle"
  ; gist:uniqueText "text/turtle"
  .
  mime:application\/trig rdf:type gist:MimeType
  ; skos:prefLabel "TriG"
  ; gist:uniqueText "application/trig"
  .
  mime:application\/n-quads rdf:type gist:MimeType
  ; skos:prefLabel "N-Quads"
  ; gist:uniqueText "application/n-quads"
  .
  mime:application\/n-triples rdf:type gist:MimeType
  ; skos:prefLabel "N-Triples"
  ; gist:uniqueText "application/n-triples"
  .
  mime:application\/rdf\+xml rdf:type gist:MimeType
  ; skos:prefLabel "RDF/XML"
  ; gist:uniqueText "application/rdf+xml"
  .
  mime:application\/json rdf:type gist:MimeType
  ; skos:prefLabel "JSON"
  ; gist:uniqueText "application/json"
  .
  mime:application\/ld\+json rdf:type gist:MimeType
  ; skos:prefLabel "JSON-LD"
  ; gist:uniqueText "application/ld+json"
  .
}
```

**Comment by Jamie-SA** [comment]
If you access the URIs you will get information about it.
Example: [https://www.iana.org/assignments/media-types/text/turtle](https://www.iana.org/assignments/media-types/text/turtle)

**Comment by sa-bpelakh** [comment]
So, when there is an externally-available URI (such as the ones which Jamie mentioned), do we want to add gist types and annotations to them (as Jamie did in the DCA), even though these URIs do not comply to our standards. Or do we want to create our own compliant URIs and then link them (via `seeAlso`, or even `sameAs`) to the external URI?

**Comment by uscholdm** [comment]
This seems like a good idea, but it should probably go into a `gistx` namespace, per convention. See issues: #206 and https://github.com/semanticarts/gist/issues/305

**Comment by Jamie-SA** [comment]
Regarding using the IANA URIs, isn't part of the promise of RDF is shared global identifiers (URIs)? This seems like a case where the URI for something exists so why wouldn't we use it.

Related is whether it should be a MIME Type or a Media Type: #434

**Comment by sa-bpelakh** [comment]
So, upon perusing #434 and the other issues mentioned (e.g. #305), I have the following PROPOSAL:
1. Rename `gist:MimeType` to `gist:MediaType`, as per #434. (The issue of `gist:GeneralMediaType`, its role in the definition of `ContentExpression` and whether it should be something like `PhysicalMediaType` is frankly out of scope, and we can happily spend a gist issue review meeting on it).
2. Define the IANA media type URIs as `gist:MediaType` instances, supplementing with any additional metadata, such as `uniqueText`, since the URIs resolve to a text document and not RDF. This way we use the URI to refer to unique real-world resources, as is the documented intention. And Jamie can remove this from the DCA ontology and inherit it directly from `gist`, and there will be much rejoicing.

Do I have consensus, @rjyounes @Jamie-SA @uscholdm , or do we need to bring this up in a larger forum?

**Comment by sa-bpelakh** [comment]
Just to add one more muddying data point, W3C refers to the Media Types formerly knowns as MIME as Internet Media Types (see https://www.w3.org/2001/tag/2004/0430-mime), so perhaps MediaType parent class, PhysicalMediaType and InternetMediaType as subclasses, and the IANA URIs as instance of InternetMediaType? Just to resolve ALL issues in one fell swoop.

**Comment by rjyounes** [comment]
While I like your proposal, I think it needs to be brought to a review session. 

If you mean PhysicalMediaType to replace our GeneralMediaType, I think the latter is far removed from what we are calling MediaType and should not be swept in with it.

**Comment by uscholdm** [comment]
> Do I have consensus, @rjyounes @Jamie-SA @uscholdm , or do we need to bring this up in a larger forum?

What about using gistx namespace for these, per #305.

**Comment by sa-bpelakh** [comment]
The whole point was to use a well known resource identifier from IANA, and not mint our own.

**Comment by uscholdm** [comment]
OK - I was not reading carefully enough, or did not fully understand what I was reading.  I was thinking something like
- `gistx:_MediaType_json`

Am I to understand that instead we will use: 
- `mime:application/json`

And if there were both, they would be connected by `owl:sameAs`?

**Comment by Jamie-SA** [comment]
Also note that creation of this data could be automated from the official IANA source data.

**Comment by Jamie-SA** [comment]
@uscholdm yes, in some serializations certain characters like slashes and '+' need to be handled properly but every RDF serialization can handle them without problem. The problems come from the human overlords.

**Comment by rjyounes** [comment]
Provisional decision:
- Separate ontology file including the IANA mime types with gist annotations
- Consult with Dave before merging PR

**Comment by rjyounes** [comment]
@sa-bpelakh Should we also consider #434 along with this issue?

**Comment by rjyounes** [comment]
Offering this proposal:

```
gist:_MediaType_TextHtml 
    a gist:MediaType; 
    gist:uniqueText "text/html";
    owl:sameAs iana:text\/html .
```
Rationale: not making assertions about a resource not in your own namespace.

**Comment by sa-bpelakh** [comment]
@rjyounes Yes, the intent is to replace `gist:MimeType` with `gist:MediaType` to be more compliant with W3C and IANA standards, as per #434

**Comment by uscholdm** [comment]
@rjyounes 
> Rationale: not making assertions about a resource not in your own namespace. 

That begs the question: what is the rationale for not making assertions about a resource not in your own namespace?  It could be that the rationale is sound based on certain reasonable assumptions which make sense in some contexts. If the context we operate in is completely outside that context (which I think it is), then there the rationale does not apply. In that case we can add assertions to concepts in other namespaces with impunity.

The advantage is simplicity and performance and not having to ever traverse an `owl:sameAs`

**Comment by rjyounes** [comment]
This is the comment thread from Teams:

Rebecca: I want to clarify the concern I raised during the discussion of mime types at the gist meeting today, because I ended up not saying what I meant. Assume for a moment we're dealing with Linked Open Data on the Semantic Web. In this context, it is problematic to use someone else's resource (i.e., a resource in a namespace you do not own) as the subject of a triple as opposed to an object. Only the owner of a namespace is able to publish data about things in that namespace, whether as RDF or HTML. Let's say I have a resource `https://mydomain.org/MarkTwain`. If someone sees that URI on a web page, and it dereferences properly, they can click on it and get to a web page that I publish with more information about the resource, possibly with links to other resources either in my domain or other people's. (The URL of the web page could be the same as the URI or different.) The same is true of RDF. Starting from my resource, you can get to all the triples I have published about it. Now let's say you have a resource `https://yourdomain.org/TomSawyer`. You can assert `https://mydomain.org/MarkTwain ex:isAuthorOf https://yourdomain.org/TomSawyer` (this is where the RDF slogan 'anyone can say anything about anything' comes in), but you will not be able to make that triple available to anyone else because you can't publish it. If I decide to ingest your triple, I can publish it, but otherwise no one will ever see it (except you, which isn't the point of LOD). This is why inverses are so important in the LOD context: you can say - and _publish_ - the triple `https://yourdomain.org/TomSawyer ex:hasAuthor https://mydomain.org/MarkTwain`. Now that triple becomes available in the LOD cloud to anyone who stumbles across it, so to speak. This is the paradigm of Linked Open Data. Another way to solve this problem is with `owl:sameAs`. You can say (and publish) `https://yourdomain.org/MarkTwain owl:sameAs https://mydomain.org/MarkTwain ; ex:isAuthorOf https://yourdomain.org/TomSawyer`. However, when it comes to vocabularies one has to be careful about using the correlates `owl:equivalentClass`  and `owl:equivalentProperty`. The authors of both FOAF and Schema agreed that `foaf:Person owl:equivalentClass schema:Person`. It would have been possible, but very bad manners, for FOAF to publish this triple without prior agreement from Schema.

 

Why is this relevant when we're not in the world of Linked Open Data, but rather in the enterprise world of proprietary data? I'm not entirely sure, but my gut feeling is that when we start creeping into the world of open data (as with the IANA mime types) we should operate according to its principles. This could be unnecessarily purist. (A similar question is 'Why do we use HTTP URIs when we are not publishing our data over HTTP?' I don't know the answer to that either. Frank at EA made the interesting point that if URIs aren't dereferenceable over HTTP then we should use URNs, to avoid giving the impression that they are.)

Boris: So, trying to avoid the bad manners of using someone else's concept as a subject (as opposed to an object) of a triple, would your proposal be:
```
gist:_MediaType_TextHtml 
    a gist:MediaType; 
    gist:uniqueText "text/html";
    owl:sameAs iana:text\/html .
```
Rebecca: Yes, that's my proposal.

Boris chatted about this later:

Boris:  to each IANA URI I added a type (`gist:MediaType`), a `prefLabel` and a `gist:uniqueText` consisting of the official media type string, (e.g. text/html).

Rebecca: Adding a type is going beyond annotations. Admittedly, using a `sameAs` and then adding a type amounts to the same thing.

Since minting our own terms depends on using `sameAs` to the IANA URI, it's effectively the same as simply using the IANA URI and adding the type and annotation properties. It then boils down to the question of whether we want to stand on the nicety of not making statements about subjects we don't own.

**Comment by uscholdm** [comment]
> And, as @uschold validly points out, we are not operating in the context of LOD, so his criteria of simplicity and not traversing a `sameAs` are valid and we should proceed with the agreed-upon implementation of using the IANA URI directly.

Yet, @rjyounes Rebecca's well-described scenario might one day be more relevant if a giant corporation like Morgan Stanley went full Enterprise Semantic Web. We might be asking the exact same question. But it might be more than a mere nicety, it might enter into the realm of political turf wars. It is something we should be thinking about a prepare for.

**Comment by sa-bpelakh** [comment]
Closed in #477
